### PR TITLE
Collection of messages to render on frontpage

### PR DESCRIPTION
**Pull request by bear454**
Instead of continuously adding sections to the frontpage notes
via settings, the collection of FrontpageMessages will be rendered
instead.

FrontpageMessages are only editable through the admin interface.

FrontpageMessages supercede the HOST.payment_statement and
HOST.use_statement settings - use of these settings will result
in a WARNING via Django checks.

### CODE REVIEW

**Comment by bear454** [comment]
@rjschwei has some additional text changed that need to be included before merge.

**Comment by rjschwei** [comment]
At the bottom of the page it should read as follows as an unmutable message:

License
openbare is licensed under the GPLv3 license. Please file issues or send contributions on GitHub https://github.com/openbare/openbare

**Comment by rjschwei** [comment]
Changes LGTM

File: library/templates/library/home.html [hunk 1]
```diff
@@ -52,18 +52,11 @@
           <dd>
             This openbare instance is operated by
             <a href="{{ host.url }}" target="_blank">{{ host.name }}</a>.
```
Comment (rjschwei) [inline library/templates/library/home.html]: Remove code above up to "Welcome"

File: library/models.py [hunk 2]
```diff
@@ -234,3 +234,17 @@ def _validate_username(self):
...
+        help_text='Messages are ordered by rank when listed on the front page',
+        blank=False
+    )
+    title = models.CharField(max_length=254, blank=False)
+    body = models.TextField(blank=False)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    class Meta:
+        ordering = ['rank', '-updated_at']
```
Comment (smarlowucf) [inline library/models.py]: Index the columns used for ordering for more performant queries. `db_index=True`

Comment (smarlowucf) [inline library/models.py]: `updated_at = models.DateTimeField(auto_now=True, db_index=True)`

File: library/models.py [hunk 3]
```diff
@@ -234,3 +234,17 @@ def _validate_username(self):
...
+class FrontpageMessage(models.Model):
+    rank = models.IntegerField(
+        default=0,
+        help_text='Messages are ordered by rank when listed on the front page',
+        blank=False
+    )
+    title = models.CharField(max_length=254, blank=False)
+    body = models.TextField(blank=False)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
```
Comment (smarlowucf) [inline library/models.py]: Was your question about model history regarding these messages? If so the history just stores that changes were made but not the historical values. If we want a history of the message (title, body) then a plugin would be needed or adding another FrontpageMessageHistory model...

Comment (bear454) [inline library/models.py]: Yes, that's something we will need to track. Thanks

File: library/models.py (hunk 2 above)
Comment (bear454) [inline library/models.py]: Duh. Adding...

File: library/templates/library/home.html (hunk 1 above)
Comment (bear454) [inline library/templates/library/home.html]: Updated in e149404.

File: library/models.py (hunk 2 above)
Comment (bear454) [inline library/models.py]: Resolved in d66cd0b.

**Comment by rjschwei** [comment]
@smarlowucf anything else?

**Comment by bear454** [comment]
@rjschwei you wanted content changes logged as well, right?

@smarlowucf says I need to add some code  - that the admin log isn't capturing the attribute changes.

**Comment by bear454** [comment]
TODO: package https://github.com/treyhunner/django-simple-history

**Comment by smarlowucf** [comment]
Would suggest using a [data migration](https://docs.djangoproject.com/en/1.10/topics/migrations/#data-migrations) for the messages rather than json fixture. They both have positives and negatives but the migration would be better to prevent accidentally overwriting data. The migrations are tracked by Django and only run once. They can also be written with reverse migrations if something happens to back up and clean things up.

**Comment by bear454** [comment]
Since this is a new table, there's no possibility of overwriting existing data. That being said, I don't expect that in all cases would the boilerplate messages be applied; in our case in particular one needs to be edited - as a migration you have no choice; as a fixture, each deployment can choose to prepopulate the table.

**Pull request merged**